File: Queue/BOJ_10845/BOJ_10845.java
import java.util.*;

public class BOJ_10845 {
  public static void main(String[] args){
    Scanner scin = new Scanner(System.in);
    int count = scin.nextInt();

    Queue qu = new Queue();

    qu.enQueue(3);

    qu.enQueue(4);

    qu.printQueue();
    // for(int i=0; i<count; i++){
    //   String order = scin.next();
    //   if (order.equals("pop")) {
    //
    //   } else if (order.equals("size")) {
    //
    //   } else if (order.equals("empty")) {
    //
    //   } else if (order.equals("front")) {
    //
    //   } else if (order.equals("back")) {
    //
    //   } else {
    //     // push
    //
    //   }
    // }
  }
}
class Node {
  int data;
  Node link;
}
class Queue {
  Node head;
  Node tail;

  Queue(){
    this.head = null;
    this.tail = null;
  }

  boolean isEmpty(){
    if (this.head == null){
      return true;
    } else return false;
  }
  void enQueue(int item){
    Node newNode = new Node();
    newNode.data = item;
    if (isEmpty()){
      head = newNode;
      tail = newNode;
    } else {
      this.tail.link = newNode;
      this.tail = newNode;
    }
  }
  int deQueue(){
    if(isEmpty()){
      return -1;
    } else {
      int item = this.head.data;
      this.head = this.head.link;
      return item;
    }
  }

  void printQueue(){
    Node temp = new Node();
    temp = this.head;

    while(temp.link != null) {
      System.out.print(temp.data + " -> ");
      temp = temp.link;
    }
    System.out.print(temp.data);
  }

}

File: BOJ/Swift/BOJ_10845.swift
import Foundation

struct Queue {
    fileprivate var dataArr: [Int] = []
    fileprivate var head = 0
    fileprivate var tail = 0

    public var size: Int {
        return dataArr.count
    }
    
    public var isEmpty: Bool {
        return dataArr.isEmpty
    }
    public var front: Int {
        if self.isEmpty {
            return -1
        }
        return dataArr[head]
    }
    public var back: Int {
        if self.isEmpty {
            return -1
        }
        return dataArr[tail-1]
    }

    mutating func enQueue(_ input: Int) {
        dataArr.append(input)
        tail += 1
    }
    mutating func deQueue() -> Int {
        if self.isEmpty {
            return -1
        }
        let result = dataArr[head]
        dataArr.remove(at:head)
        tail -= 1
        
        return result
    }
}

var q = Queue()


if let number = Int(readLine()!) {
    for i in 1...number {
        if let input = readLine() {
            let inputArr : [String] = input.components(separatedBy: " ")
            
            switch inputArr[0] {
            case "push":
                q.enQueue(Int(inputArr[1])!)
            case "pop":
                print(q.deQueue())
            case "front":
                print(q.front)
            case "back":
                print(q.back)
            case "size":
                print(q.size)
            case "empty":
                print(q.isEmpty ? 1 : 0)
            default:
                fatalError()
                break
            }
            
        }
    }
}
